std/hash/poseidon_bn254.asm

```diff
@@ -65,21 +65,18 @@ machine PoseidonBN254(LASTBLOCK, operation_id) {
     pol commit input_state[STATE_SIZE];
 
     // Add round constants
-    pol commit a[STATE_SIZE];
-    array::zip3(a, state, C, |a, state, C| a = state + C);
+    let a: expr[STATE_SIZE] = array::zip(state, C, |state, C| state + C);
 
     // Compute S-Boxes (x^5)
-    pol commit x2[STATE_SIZE], x4[STATE_SIZE], x5[STATE_SIZE];
-    array::zip2(x2, a, |x2, a| x2 = a * a);
-    array::zip2(x4, x2, |x4, x2| x4 = x2 * x2);
-    array::zip3(x5, x4, a, |x5, x4, a| x5 = x4 * a);
+    let x2: expr[STATE_SIZE] = array::map(a, |a| a * a);
```

Comment:
From the perspective of witgen, the two are equivalent, as intermediate columns are inlined anyway. But we do pass something different to the backend, and in fact, eStark fails if we pass the inlined version.

We could automate creating these intermediate polynomials. But it's not trivial, because I guess computing `(x * x) * (x * x)` is more efficient than `(((x * x) * x) * x)`.